kong/tracing/propagation.lua

```diff
@@ -41,7 +42,7 @@ local function parse_baggage_headers(headers, header_pattern)
   -- account for both ot and uber baggage headers
   local baggage
   for k, v in pairs(headers) do
-    local baggage_key = match(k, header_pattern)
+    local baggage_key = find(k, header_pattern, 1, true)
```

Comment:
This doesn't seem quite right. Using `string.find` this way performs a plain substring search, but the [possible inputs](https://github.com/Kong/kong/blob/8910980b2dcc580c6cd93130799b06115a6b853f/kong/tracing/propagation.lua#L21-L22) are all patterns and not plain strings. Also, `string.find` returns the position of the match, but the surrounding context makes it clear that we are interested in the matched substring itself (hence `string.match` being used).

Reply:
fixed